src/sdk/sql_cluster_router.cc

```diff
@@ -1549,5 +1550,83 @@ std::shared_ptr<hybridse::sdk::Schema> SQLClusterRouter::GetTableSchema(
     return {};
 }
 
+std::vector<std::string> SQLClusterRouter::ExecuteDDLParse(
+    const std::string& sql,
+    const std::vector<std::pair<std::string, std::vector<std::pair<std::string, hybridse::sdk::DataType>>>>&
+        table_map) {
+    using openmldb::base::DDLParser;
+    std::map<std::string, std::vector<openmldb::common::ColumnDesc>> table_desc_map;
+    for (const auto& table_item : table_map) {
+        std::string table_name = table_item.first;
+        std::vector<std::pair<std::string, hybridse::sdk::DataType>> column_list = table_item.second;
+        std::vector<openmldb::common::ColumnDesc> column_desc_list;
+        for (const auto& column_map : column_list) {
+            openmldb::common::ColumnDesc column_desc;
+            std::string column_name = column_map.first;
+            hybridse::sdk::DataType column_type = column_map.second;
+            column_desc.set_name(column_name);
+            openmldb::type::DataType data_type;
+            if (!openmldb::catalog::SchemaAdapter::ConvertType(column_type, &data_type)) {
+                return {};
+            }
+            column_desc.set_data_type(data_type);
+            column_desc_list.push_back(column_desc);
+        }
+        table_desc_map.insert(std::make_pair(table_name, column_desc_list));
+    }
+
+    openmldb::base::IndexMap index_map = openmldb::base::DDLParser::ExtractIndexes(sql, table_desc_map);
+    std::vector<std::string> ddl_vector;
+    for (const auto& table_item : table_desc_map) {
+        std::string ddl = "CREATE TABLE IF NOT EXISTS ";
+        std::string table_name = table_item.first;
+        ddl = ddl.append(table_name);
+        ddl = ddl.append("(\n");
+        DLOG(INFO) << "table_name is " + table_name;
+        auto column_desc_list = table_item.second;
+        for (const auto& column_desc : column_desc_list) {
+            const auto column_name = column_desc.name();
+            const auto data_type = column_desc.data_type();
+            std::string column_type = openmldb::codec::DATA_TYPE_STR_MAP.find(data_type)->second;
+            ddl = ddl.append("\t");
+            ddl = ddl.append(column_name);
+            ddl = ddl.append(" ");
+            ddl = ddl.append(column_type);
+            ddl = ddl.append(",\n");
+        }
+        std::string index_table = index_map.find(table_name)->first;
```

Comment:
index_map.find(table_name) may get invalid iterator, you can't get first.
fix e.g.
```
auto iter = index_map.find(table_name);
if(iter!=index_map.end()){
...
}
```
And it's better to warning when `iter==index_map.end()`